### ISSUE TITLE
Stop parsing when cell x has value y

### ISSUE THREAD

**Issue by jacobduijzer** [josdeweger-zz/SheetToObjects#47]
I am importing product data but when importing I do not know the number of rows so I provide a large value (for Excel), e.g. 100000 rows. It takes a serious amout of time to parse the document while there are only 10 rows available. 

It should be nice to have a StopParsingRule, e.g.:

```
var sheetMapper = new SheetMapper()
                .AddConfigFor<AdditionExcelImport>(cfg => cfg
.MapColumn(column => column
                        .WithHeader("Code")
                        .MapTo(m => m.Code)
.StopParsingWhen(x => x == null)
)

```

I might try and implement something like this but would like to know if anyone else have an interest in this feature.

**Comment by josdeweger-zz** [comment]
Hi @jacobduijzer, as a first version, something like this #48?
This is lightweight, because it will simply filter out all the data after the first empty row, before starting to parse it.

**Comment by jacobduijzer** [comment]
Hi @josdeweger. Reviewed and approved your solution but it's not the solution I was looking for. With your solution the max number of rows will be read before processing (and potentially stopping when an empty row is seen). I have just created  PR #49 which will stop reading the file completely on the first empty row. This shows a serious drop in memory usage when reading the max possible lines from an excel sheet.

**Comment by josdeweger-zz** [comment]
Cool stuff, will close this as it is working as expected
